src/components/common/DropdownButton.js

```diff
@@ -1,77 +1,82 @@
+    displayValue,
+    displayClass,
+    toggleClass,
+    toggleColor,
+    toggleSize,
+  } = props;
 
-  //= =============React Lifecycle Functions===================//
-  componentDidMount() {}
+  const parentDisplay = displayClass + '-language-dropdown';
+  const itemDisplay = displayClass + '-language-dropdown-closed-content';
+  const [dropdownOpen, setdropdownOpen] = useState(defaultOpen || false);
 
-  toggleHandler = (prevVal) => {
-    this.setState({ dropdownOpen: !prevVal });
+  const toggleHandler = (prevVal) => {
+    setdropdownOpen(!prevVal);
   };
 
-  selectLanguage = (program) => {
-    let result = true;
-    if (this.props.dirty) {
-      result = window.confirm('Are you sure you want to change programs? You have unsaved changes');
-    }
+  const renderDropdownItems = () =>
+    children.map(({ display, value }) => {
+      return (
+        <DropdownItem key={value} onClick={() => onSelect({ display, value, dirty })}>
+          {display}
+        </DropdownItem>
+      );
+    });
 
-    if (this.props.onSelect && result) {
-      this.props.onSelect(program);
-    }
-  };
-
-  renderDropdownItems = () =>
-    // map each program string in the array to a dropdown item
-    this.props.dropdownItems.map((program) => (
-      <DropdownItem key={program.key} onClick={() => this.selectLanguage(program.key)}>
-        <FontAwesomeIcon icon={program.language.icon} fixedWidth />
-        <span style={{ marginLeft: '10px' }}>{program.name}</span>
-      </DropdownItem>
-    ))
-  ;
-
-  render() {
-    // let value = this.props.displayValue
-    // if(this.props.dirty){
-    //   value = (<span>&#8226;{this.props.displayValue}</span>)
-    // }
+  return (
+    <div className={parentDisplay}>
+      <Dropdown isOpen={dropdownOpen} toggle={() => toggleHandler(dropdownOpen)}>
+        {/* HACK: disables the colors entirely, makes the dropdown transparent*/}
+        <DropdownToggle className={toggleClass} color={toggleColor} size={toggleSize} caret>
+          <div className={itemDisplay}>{displayValue}</div>
+        </DropdownToggle>
+        <DropdownMenu>{renderDropdownItems()}</DropdownMenu>
+      </Dropdown>
+    </div>
+  );
+};
+export default DropdownButton;
 
-    return (
-      <div className="editor-language-dropdown">
-        <Dropdown
-          isOpen={this.state.dropdownOpen}
-          toggle={() => this.toggleHandler(this.state.dropdownOpen)}
-        >
-          {/* HACK: disables the colors entirely, makes the dropdown transparent */}
-          <DropdownToggle className="btn-language-dropdown" color="" caret>
-            <div className="editor-language-dropdown-closed-content">
-              <FontAwesomeIcon icon={this.props.currentLanguage.icon} fixedWidth />
-              {' '}
-              {this.props.displayValue}
-            </div>
-          </DropdownToggle>
-          <DropdownMenu>{this.renderDropdownItems()}</DropdownMenu>
-        </Dropdown>
-      </div>
-    );
-  }
-}
+/*
```

Comment:
Obviously these should eventually be removed, but I understand having it here for reference.